### ISSUE TITLE
(xml) XML/HTML parsing is error prone & not able to input XML unparsed

### ISSUE THREAD

**Issue by Yenthe666** [highlightjs/highlight.js#1778]
Hi guys,

First of all: a huge thank you for this library! It looks great and works fine out of the box for plenty of languages, such as Python.
With XML however I have a problem. Ideally I'd like to input the raw XML just as it is, so like:
```
<span class="hljs-meta"><!DOCTYPE html></span>
```
Instead of:
```
<span class="hljs-meta">&lt;!DOCTYPE html&gt;</span>
```
Is there any way to do this? I've tried to wrap it in `CDATA` for example to make sure it doesn't get parsed but that doesn't seem to work.
So then I decided that encoding was ok for now. This is where things got weird. Have a look at this line of code:
```
&lt;field name
```
If I render this it will not print out anything, while if I place a space between &lt; and the "f" from field name it will, this is the code:
```
&lt; field name
```

If I add them both under eachother like this:
```
&lt; field name
&lt;field name
```
Then the first one is printed in the webpage, the second one is completely ignored and not rendered:
![image](https://user-images.githubusercontent.com/6352350/43357673-934c01fe-9285-11e8-9ee1-924c7210b5de.png)
When I change this order around like this:
```
&lt;field name
&lt; field name
```
Nothing will be rendered for both items:
![image](https://user-images.githubusercontent.com/6352350/43357677-b2231e46-9285-11e8-9bae-84dbc71647b7.png)

Then you might say okay it is not working because the XML is not correctly closed off. But this one doesn't work either:
```
&lt;field name&#61;&quot;description&quot;&gt;&lt;&#47;field&gt;
```

So I have two questions:
- Is there a way to load pure XML in the `<pre><code>` blocks and if so, how?
- Why do so many statements fail for conversion?

**Comment by yyyc514** [comment]
> Is there any way to do this? I've tried to wrap it in CDATA for example to make sure it doesn't get parsed but that doesn't seem to work.

CDATA is an XML thing, not an HTML thing.  While it may once have been valid in the age of XHTML it's no longer a thing.  There is no great way to dump unescaped HTML into an HTML file.

> Is there a way to load pure XML in the <pre><code> blocks and if so, how?

So, no. If you were getting the data from some other place (AJAX, etc) you could perhaps build a pipeline for this yourself with javascript... Or you could even encode the data in JS instead:

```
var code = "<html><blah>"
```

But then you still have to escape it, just using Javascript's rules about quoting, not HTML's more complex rules...  If you just want to embed it inside the <code> tag though then yes, it must be escaped properly.

I find it hard to follow the rest though, but I'm not even going to try without an actual example I can confirm and play around with (since this is so nuanced).  If there is an a issue here (other than the encoding one, which I've addressed) please create a jsfiddle to show working examples of the issue and I'll have a look.

You can even use my template:
https://jsfiddle.net/ajoshguy/nagkqytv/18/

**Comment by yyyc514** [comment]
Weekend going.  Is there still an issue here?

**Comment by Yenthe666** [comment]
Thanks for the detailed feedback and info. I've accepted the fact that there's not much that can be done to change it without going with JS functions or something equivalent. I've made workarounds to enable this so I guess that's just the way it works and how it should. :)
Dont worry too much about this one, I assume you're completely right. Thanks for the feedback.

**Comment by yyyc514** [comment]
A script tag might be better (for raw content) but then you’d need custom JS to move the content into a code block and call Highlight.js.  Might not be hard to pair them with auto generated ids or something if your generating the content programatically. 

Is writing a little custom JS not an option?

**Comment by Yenthe666** [comment]
I ended up with doing a combination of both. It's not a perfect solution but it works OK, just had to add some extra JS.

**Comment by yyyc514** [comment]
Awesome.
